packages/@o3r/store-sync/src/core/storage-sync.ts

```diff
@@ -1,22 +1,35 @@
-import { INIT, UPDATE } from '@ngrx/store';
-import { deepFill } from '@o3r/core';
-import type { StorageSyncOptions } from './interfaces';
-import { isLocalStorageConfig, isSerializer, rehydrateAction } from './storage-sync.helpers';
-import { syncStorage } from '../sync-storage';
+import {INIT, UPDATE} from '@ngrx/store';
+import {deepFill} from '@o3r/core';
+import equal from 'fast-deep-equal';
+import type {StorageSyncOptions} from './interfaces';
+import {isLocalStorageConfig, isSerializer, rehydrateAction} from './storage-sync.helpers';
+import {syncStorage} from '../sync-storage';
+import {StorageSyncConstructorOptions} from './interfaces';
 
 /**
  * Storage synchronizer
  */
 export class StorageSync {
   private alreadyHydratedStoreSlices: Set<string> = new Set();
   private hasHydrated = false;
+  private storeImage: { [key: string]: any } = {};
 
   public options: StorageSyncOptions;
 
-  constructor(options?: Partial<StorageSyncOptions>) {
+  constructor(options?: StorageSyncConstructorOptions, extraOptions?: {disableSmartSync: boolean}) {
     this.options = {
       keys: [],
+      ...(extraOptions?.disableSmartSync ? {} : {
+        syncKeyCondition: (key, state) => !equal(this.storeImage[key], state[key]),
```

Comment:
I don't know how easy it would be to implement but it could be nice to have some kind of `throttle` (with configurable delay maybe?) to generally avoid any unnecessary CPU overhead